content/the-basics-of-go.md

```diff
@@ -0,0 +1,865 @@
+files within a single folder. By convention, the main entry point for Go program will be called package main. Whereas
+for other packages, it will be called as the directory name. So that if I happen to have a folder called "controllers"
+that lives as a subfolder from the main package, it will be named as package controllers. Which shaped this folder
+structure:
+
+```
+.
+├── main.go       (package main)
+└── controllers
+    ├── index.go  (package controllers)
+    └── users.go  (package controllers)
+```
+
+Consider the directory and package structure above. The first question that I had when learning about Go is: how do you
+actually import a Go file?
+
+Well, you don't. You don't import a file, because this is not a scripting language like Javascript, PHP, or Python. If you
+want to import a file, you should import the whole package with the keyword `import "<package name>"`.
+
+If I have some function that I want to import from the `controllers/index.go` into the `main.go` file, I would do:
+
+```go
+// main.go
+package main
+
+import "project/controllers"
+
+func main() {
+  controllers.SayHello()
+}
+
+// controllers/index.go
+package controllers
+
+import "fmt"
+
+func SayHello() {
+  fmt.Println("Hello!")
+}
+```
+
+Simple, right? It also goes the same for things that are on the `controllers/users.go`, you just need to import
+`controllers` once.
+
+What about public and private scope in Go?
+
+It's straightforward:
+
+- Anything that would meant to be public scope (a type, function, variable, or constant) should be named with an uppercase
+  on the first character. Example: `func SayHello()`.
+- Anything that would meant to be a private scope, should do the opposite of public scope. It should be named with a lowercase
+  on the first character. Example: `func sayHello()`.
+
+## Data types
+
+Go has some built-in data types, which I will only mention some of the important ones:
+
+- `bool` - boolean value being either true or false.
+- `string` - a string of characters.
+- `byte` - byte is an alias for uint8 and is equivalent to uint8 in all ways. It is used, by convention,
+  to distinguish byte values from 8-bit unsigned integer values.
+- `error` - error built-in interface type is the conventional interface for representing an error condition, with the nil value representing no error.
+- `int` - int is a signed integer type that is at least 32 bits in size. It is a distinct type, however,
+  and not an alias for, say, int32.
+- `uint` - uint is an unsigned integer type that is at least 32 bits in size. It is a distinct type, however, and not an alias for, say, uint32.
+
+## Variables
+
+There are some way to create a variable in Go:
+
+```go
+package main
+
+func main() {
+  // Oh by the way, this is a comment.
+  var str string // Declares a variable called str, which holds an empty string
+  // Due to the is equivalent to:
```

Comment:
```suggestion
  // The declaration above is equivalent to:
```

Reply:
I was wondering why I wrote this line in the first place yesterday lol, but ok